src/activities/binary_bulb/BinaryBulb.qml

```diff
@@ -0,0 +1,200 @@
+
+        IntroMessage {
+            id: message
+            onIntroDone: {
+                Activity.initLevel()
+            }
+            intro: [
+                qsTr("Computers use Binary number system, where there are two symbols, 0 and 1."),
+                qsTr("In decimal number system 123 is represented as 1 x 100 + 2 x 10 + 3 x 1"),
+                qsTr("Binary represents numbers in the same pattern, but using powers of 2 instead of powers of 10 that decimal uses"),
+                qsTr("So, 1 in binary is represented by 001, 4 by 100, 7 by 111 and so on.."),
+                qsTr("Our computer has many many switches (called transistors) that can be turned on or off given electricity, and a switch that is on will represent a 1 and a switch that is off will represent a 0."),
+                qsTr("In this activity, you are given a number, you have to find its binary representation by turning on the bulbs. An on bulb representes 1 and an off bulb represents 0")
+            ]
+            z: 20
+            anchors {
+                top: parent.top
+                topMargin: 10
+                right: parent.right
+                rightMargin: 5
+                left: parent.left
+                leftMargin: 5
+            }
+        }
+
+        Rectangle {
+            id: questionItemBackground
+            opacity: 0.00
+            z: 10
+            anchors {
+                horizontalCenter: parent.horizontalCenter
+                bottomMargin: 10
+            }
+            width: parent.width - 20
+            height: parent.height * 0.30
+        }
+
+        GCText {
+            id: questionItem
+            anchors.fill: questionItemBackground
+            anchors.bottom: questionItemBackground.bottom
+            fontSizeMode: Text.Fit
+            fontSize: largeSize
+            wrapMode: Text.Wrap
+            z: 10
+            color: "white"
+            verticalAlignment: Text.AlignVCenter
+            horizontalAlignment: Text.AlignHCenter
+            text: qsTr("What is the binary representation of %1 ?").arg(items.numberToConvert)
+        }
+
+        Row {
+            id: row
+            anchors.centerIn: parent
+            spacing: 10 * ApplicationInfo.ratio
+            focus: true
+            Repeater {
+                id: bulbs
+                model: items.numberOfBulbs
+                LightBulb {
+                    height: background.height / 5
+                    width: (background.width > background.height) ? (background.width / 20) : ((background.width - (16 * row.spacing)) / 8)
+                }
+            }
+        }
+
+        GCText {
+            id: reachedSoFar
+            anchors.horizontalCenter: row.horizontalCenter
+            anchors.top: row.bottom
+            anchors.topMargin: 25
+            color: "white"
+            fontSize: largeSize
+            text: items.numberSoFar
+            visible: items.score.currentSubLevel <= 2 ? true : false
```

Comment:
I think it would be better to have separate levels for it.
Odd levels with the texts and even without (but with the same number of bulbs than the previous level).
So level 1, 2 bulbs and text, level 2, 2 bulbs no text, level 3, 4 bulbs, text...